stag-library-compiler/src/main/java/com/vimeo/stag/processor/generators/AdapterGenerator.java

```diff
@@ -58,16 +57,26 @@ static String getJsonName(@NotNull Element element) {
     }
 
     /**
-     * Returns the alternate name for the {@link Element}
+     * Returns the alternate names for the {@link Element}.
+     *
+     * @param element the element to check for alternate names.
+     * @return an array of alternate names, or null if there are none.
      */
     @Nullable
     static String[] getAlternateJsonNames(@NotNull Element element) {
-        return (null != element.getAnnotation(SerializedName.class)) ? element.getAnnotation(
-                SerializedName.class).alternate() : null;
+        return null != element.getAnnotation(SerializedName.class) ?
+                element.getAnnotation(SerializedName.class).alternate() :
+                null;
     }
 
+    /**
+     * Creates a {@link TypeSpec} for the the type adapter.
+     *
+     * @param typeTokenConstantsGenerator the generator for the TypeTokenConstants class.
+     * @param stagGenerator               the generator for the Stag class.
+     * @return A non null {@link TypeSpec} describing the adapter class.
+     */
     @NotNull
-    public abstract TypeSpec getTypeAdapterSpec(
-            @NotNull TypeTokenConstantsGenerator typeTokenConstantsGenerator,
-            @NotNull StagGenerator stagGenerator);
+    public abstract TypeSpec createTypeAdapterSpec(@NotNull TypeTokenConstantsGenerator typeTokenConstantsGenerator,
```

Comment:
Okay so I went through the code and it looks like the `TypeTokenConstantsGenerator` code should never be called. This is because in all the cases where it would have been called it would have been called in a case where a type was not a generic type (`TypeKind.TYPEVAR`) or did not contain any unresolved generic types in its declaration (e.g. `HashMap<String, T>`), but everywhere it was called was protected by either `TypeUtils.isConcreteType` or `TypeUtils.containsTypeVarParams` checks that would functionally prevent those branches from ever being hit. What I did was remove all the unnecessary code and add in preconditions checks to ensure that the conditions that are met right now in the code are reconsidered before the code is misused elsewhere. I also renamed some methods to specify that they are for generic types only.

See this commit: https://github.com/vimeo/stag-java/pull/77/commits/2a32acdf44a7ceb8b90cd69766e099cfad1863a2